app/workers/delete_plain_object_worker.rb

```diff
@@ -9,10 +9,10 @@ class DeletePlainObjectWorker < ActiveJob::Base
   end
 
   rescue_from(ActiveRecord::RecordNotDestroyed, ActiveRecord::StaleObjectError) do |exception|
-    record = exception.record
-    if record.class.exists?(record.id)
+    if object.class.exists?(object.id)
       System::ErrorReporting.report_error(exception, parameters: { caller_worker_hierarchy: caller_worker_hierarchy,
-                                                                   error_messages: record.errors.full_messages })
+                                                                   error_messages: exception.record.errors.full_messages })
+      retry_job
```

Comment:
Probably there is another way to do it. Anyway it deserves an explanation:

The issue here is a race condition. A child object was loaded but concurrently deleted by another worker.
Thus, when trying to delete the child object (through `dependent: :destroy`) it will raise `ActiveRecord::StaleObjectError`

Could you add a comment, otherwise it will be cryptic or separate the rescue_from in 2 cases

```ruby
rescue_from(ActiveRecord::RecordNotDestroyed) do |exception|
  # Some logs to explain why we skip retrying the deletion here
end

rescue_from(ActiveRecord::StaleObjectError) do |exception|
  # the code for this case
end
```

Reply:
Makes sense 👍 will do 👍